components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -56,13 +57,15 @@ type gpsrtk struct {
 	// everything below this comment is protected by mu
 	ntripClient      *gpsutils.NtripInfo
 	cachedData       *gpsutils.CachedData
-	correctionWriter io.ReadWriteCloser
+	correctionWriter io.WriteCloser
 	writePath        string
 	wbaud            int
 	isVirtualBase    bool
-	readerWriter     *bufio.ReadWriter
-	writer           io.Writer
-	reader           io.Reader
+	vrsReaderWriter  *bufio.ReadWriter
+	vrsConn          net.Conn
```

Comment:
added the vrs connection to the struct so we can close the connection when the resource gets closed